### PR TITLE
Preparation for 3.8.0.20220526 pre-release up to and including release notes

### PR DESCRIPTION
**Pull request by Mikolaj**
This is done by tracking the checklists at https://github.com/haskell/cabal/wiki/Making-a-release

Let's (revise and) merge this phase before we tackle the next (in particular, bootstrap and GHA and gitlab CI changes; I wonder if CI failes due to not being updated to new version on this branch; or perhaps it's going to pass voidly, testing a previous version).

### CODE REVIEW

**Comment by Mikolaj** [comment]
Actually, the abruptness with which CI fails surprises me. What's going on? What should I update in GHA setup? Or should something in the github UI setting tab be changed?

**Comment by Mikolaj** [comment]
CI seems stuck; let me push -f.

**Comment by Mikolaj** [comment]
Hmm, it seems only the display was stuck, despite reloading the page. Now CI seems green for the previous push. \o/

**Review by Kleidukos — CHANGES_REQUESTED** [review]
no wait actually, I think the "BSD3" symbol is not a valid SPDX identifier. I think we're looking for BSD-3-Clause

**Comment by Kleidukos** [comment]
> > no wait actually, I think the "BSD3" symbol is not a valid SPDX identifier. I think we're looking for BSD-3-Clause
> 
> Oh, nice catch. You mean, in the cabal files where I bumped cabal-version, I now need to adhere to SPDX as well, right? Will do.

Yeah, even if Cabal syntax version 2.0 doesn't complain, it's better to be future-proof

**Comment by Mikolaj** [comment]
@Kleidukos: done in #8160. Will be backported (with lots of conflicts).

**Comment by Mikolaj** [comment]
It turns out the "stuck CI" on 3.8 branch was just the outdated 3.* projection rules automatically applying to 3.8. I've updated them.

**Review by Kleidukos — APPROVED** [review]
beautiful

**Review by ulysses4ever — APPROVED** [review]

File: cabal-testsuite/cabal-testsuite.cabal [hunk 1]
```diff
@@ -111,6 +111,8 @@ executable setup
 
 custom-setup
   -- we only depend on even stable releases of lib:Cabal
-  setup-depends: Cabal == 2.2.* || == 2.4.* || == 3.0.* || ==3.2.* || ==3.4.* || ==3.6.*,
+  -- and due to Custom complexity and ConstraintSetupCabalMaxVersion
+  -- it can't be the current version
+  setup-depends: Cabal == 3.6.*,
```
[the remark this answers is not in the document]
Comment (Mikolaj) [reply cabal-testsuite/cabal-testsuite.cabal:116]: Yes, I don't understand it, either, it's something about Custom setup and I don't touch it with a pole. But a code dive for the exception I was getting (`ConstraintSetupCabalMaxVersion` or something related) got me to this conclusion, hence the comment. Could the comment be improved?

[the remark this answers is not in the document]
Comment (ulysses4ever) [reply cabal-testsuite/cabal-testsuite.cabal:116]: I think it's fine. I guess I'd just make sure that it's also mentioned on the "Making the release" wiki page.

[the remark this answers is not in the document]
Comment (Mikolaj) [reply cabal-testsuite/cabal-testsuite.cabal:116]: Indeed, thank you. Added to the wiki page.

**Comment by Mikolaj** [comment]
To be honest, the change to [cabal-testsuite/src/Test/Cabal/OutputNormalizer.hs](https://github.com/haskell/cabal/pull/8163/files#diff-2d425fcad95b791b03a5353137907a1db325888f72a6dcce123d172ab3cd8e32) would normally require a changelog, but given that it's a tiny fix, made on a branch, etc., let's be forgiving.

**Pull request merged**